Source/WTF/wtf/Int128.h

```diff
@@ -1264,10 +1267,73 @@ using UInt128 = UInt128Impl;
 using Int128 = Int128Impl;
 #endif
 
+template<> struct DefaultHash<UInt128> {
+    static unsigned hash(const UInt128& i) { return pairIntHash(static_cast<uint64_t>(i >> 64), static_cast<uint64_t>(i)); }
+    static bool equal(const UInt128& a, const UInt128& b) { return a == b; }
+    static constexpr bool safeToCompareToEmptyOrDeleted = true;
+};
+template<> struct DefaultHash<Int128> {
+    static unsigned hash(const UInt128& i) { return pairIntHash(static_cast<uint64_t>(i >> 64), static_cast<uint64_t>(i)); }
+    static bool equal(const Int128& a, const Int128& b) { return a == b; }
+    static constexpr bool safeToCompareToEmptyOrDeleted = true;
+};
+
+#if !HAVE(INT128_T)
+template<> struct HashTraits<UInt128> : GenericHashTraits<UInt128> { };
+template<> struct HashTraits<Int128> : GenericHashTraits<Int128> { };
+#endif
```

Comment:
How about just always defining this regardless of whether we are using uint128_t?
```
template<> struct HashTraits<UInt128> : GenericHashTraits<UInt128> {
    static constexpr bool emptyValueIsZero = true;
    static void constructDeletedValue(UInt128& slot) { slot = static_cast<UInt128 >(-1); }
    static bool isDeletedValue(UInt128 value) { return value == static_cast<UInt128 >(-1); }
};
```
And ditto for `Int128`.